### PR TITLE
[BISERVER-14192] ASCII Control Characters are being allowed in Folder…

### PR DESCRIPTION
**Pull request by smmribeiro**
…s and File Names causing repository corruption when backing up and restoring

@pentaho-lmartins @ssamora @ppatricio

### CODE REVIEW

**Comment by buildguy** [comment]
### :warning: Build finished in 29m 31s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 6 test(s) failed:

<details><summary><b>org.pentaho.platform.plugin.services.importer.PRPTImportHandlerTest.testImportFile_slashPath</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

<details><summary><b>org.pentaho.platform.plugin.services.importer.PRPTImportHandlerTest.testImportFile_metadataIsCorrect</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

<details><summary><b>org.pentaho.platform.plugin.services.importer.PRPTImportHandlerTest.testImportFile_metadataReturnOnlyDescription</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

<details><summary><b>org.pentaho.platform.plugin.services.importer.PRPTImportHandlerTest.testImportFile_validPath</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

<details><summary><b>org.pentaho.platform.plugin.services.importer.PRPTImportHandlerTest.testImportFile_metadataReturnOnlyTitle</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

<details><summary><b>org.pentaho.platform.plugin.services.importer.PRPTImportHandlerTest.testImportFile_backSlashPath</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

  
Tests run: **1515**, Failures: **6**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4712/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 31m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1515**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4712/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 30m 16s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1515**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4712/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportFileHandler.java [hunk 1]
```diff
@@ -154,6 +155,24 @@ public void importFile( IPlatformImportBundle bnd ) throws PlatformImportExcepti
...
+    if ( null != bundle ) {
+      FileService fileService = new FileService();
+      String name = bundle.getName();
+      if ( bundle.isFolder() ) {
+        if ( !fileService.isValidFolderName( name ) ) {
+          throw new PlatformImportException( messages.getString( "DefaultImportHandler.ERROR_0012_INVALID_FOLDER_NAME",
+            name ), PlatformImportException.PUBLISH_PROHIBITED_SYMBOLS_ERROR );
+        }
+      } else {
+        if ( !fileService.isValidFolderName( name ) ) {
```
Comment (ssamora) [inline extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportFileHandler.java:168]: should be isValidFileName?

File: extensions/src/main/java/org/pentaho/platform/plugin/services/importer/SolutionImportHandler.java [hunk 2]
```diff
@@ -243,7 +197,7 @@ public void importFile( IPlatformImportBundle bundle ) throws PlatformImportExce
         bundleInputStream = new ByteArrayInputStream( bytes );
         // If is locale file store it for later processing.
         if ( localeFilesProcessor.isLocaleFile( fileBundle, importBundle.getPath(), bytes ) ) {
-          log.trace( "Skipping [" + repositoryFilePath + "], it is a locale property file" );
+          getLogger().trace( "Skipping [" + repositoryFilePath + "], it is a locale property file" );
```
Comment (ssamora) [inline extensions/src/main/java/org/pentaho/platform/plugin/services/importer/SolutionImportHandler.java:200]: Externalize all hard coded strings for i18n.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 14s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1515**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4712/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 31m 9s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1515**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4712/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 30m 28s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1515**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4712/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**